src/ngx_header_writer.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "string_util.h"
+#include <datadog/dict_writer.h>
+
+extern "C" {
+#include <nginx.h>
+#include <ngx_config.h>
+#include <ngx_core.h>
+#include <ngx_http.h>
+}
+
+namespace datadog {
+namespace nginx {
+
+class NgxHeaderWriter : public datadog::tracing::DictWriter {
+  ngx_http_request_t *request_;
+  ngx_pool_t *pool_;
+
+public:
+  explicit NgxHeaderWriter(ngx_http_request_t *request)
+      : request_(request), pool_(request_->pool) {}
+
+  void set(std::string_view key, std::string_view value) override {
+    // Question: Should we check first if the key already exist?
+    const auto key_size = key.size();
+
+    ngx_table_elt_t *h = (ngx_table_elt_t*)ngx_list_push(&request_->headers_in.headers);
+    if (h == NULL) {
+      return;
+    }
+
+    h->hash = 1;
```

Comment:
Please add a comment here explaining why (even if it's a link to some docs or a mailing list thread).